zerver/tests/test_outgoing_webhook_system.py

```diff
@@ -186,6 +187,37 @@ def test_request_exception(self) -> None:
 When trying to send a request to the webhook service, an exception of type RequestException occurred:
 ```
 I'm a generic exception :(
+```''')
+        assert bot_user.bot_owner is not None
+        self.assertEqual(bot_owner_notification.recipient_id, bot_user.bot_owner.id)
+
+    def test_jsonable_exception(self) -> None:
+        bot_user = self.example_user('outgoing_webhook_bot')
+        mock_event = self.mock_event(bot_user)
+        service_handler = GenericOutgoingWebhookService("token", bot_user, "service")
+
+        # Widget content is being decoded twice, once in process_success_response along with response
+        # and again in check_messages leading to always cause an error
+        response_body = orjson.dumps(dict(content='whatever', widget_content="test"))
+        expect_logging_exception = self.assertLogs(level="ERROR")
+        expect_fail = mock.patch("zerver.lib.outgoing_webhook.fail_with_message")
+
+        with responses.RequestsMock(assert_all_requests_are_fired=True) as requests_mock:
+            requests_mock.add(
+                requests_mock.POST,
+                'https://example.zulip.com',
+                status=200,
+                body=response_body)
+            with expect_logging_exception, expect_fail as mock_fail:
+                do_rest_call('https://example.zulip.com', None, mock_event, service_handler)
+            self.assertTrue(mock_fail.called)
+
+            bot_owner_notification = self.get_last_message()
+            # print(bot_owner_notification.content)
+            self.assertEqual(bot_owner_notification.content, '''[A message](http://zulip.testserver/#narrow/stream/999-Verona/topic/Foo/near/) triggered an outgoing webhook.
+When trying to send a request to the webhook service, an exception of type JsonableError occurred:
+```
+Widgets: API programmer sent invalid JSON content
```

Comment:
I think we want to tweak the logic to avoid exposing internal details to the user.  Specifically, it shouldn't mention `JsonableError` etc.  I think good text would be something like (changing only starting from the second line):

```
The outgoing webhook server attempted to send a message in Zulip, but that request resulted in the following error:

> Widgets: API programmer sent invalid JSON content
```

(Structurally, we're just exposing to the user the API response they'd have gotten via the Zulip `send_message` API).